frontend/src/hooks/api/getters/useSegments/useSegments.ts

```diff
@@ -16,9 +15,13 @@ export interface IUseSegmentsOutput {
 export const useSegments = (strategyId?: string): IUseSegmentsOutput => {
     const { uiConfig } = useUiConfig();
 
+    const url = strategyId
+        ? formatApiPath(`api/admin/segments/strategies/${strategyId}`)
+        : formatApiPath('api/admin/segments');
+
     const { data, error, mutate } = useSWR(
-        [strategyId, uiConfig.flags],
-        fetchSegments,
+        url,
+        () => (uiConfig.flags?.SE ? fetchSegments(url) : []),
```

Comment:
We tried this approach in change requests and it didn't work. Please check SWR recommended approach: https://swr.vercel.app/docs/conditional-fetching. We have our own hook useConditionalSWR encapsulating the logic.